### ISSUE TITLE
Dataflow Analysis adds "this.fieldValue" to initial store of a static method

### ISSUE THREAD

**Issue by AndrewShf** [eisop/checker-framework#215]
When doing dataflow analysis for a static method, it's redundant to insert `this.certain-field-value` to the initial store, as `this` refers to the current instance of the class.

Here is a minimal example that can reproduce the problem:
Running `javacheck -processor org.checkerframework.checker.nullness.NullnessChecker -Aflowdotdir=. Test.java`

Test.java is here:
```
import org.checkerframework.checker.nullness.qual.NonNull;
import org.checkerframework.checker.nullness.qual.Nullable;

class Test {
    @Nullable Object f;
    static void test() {
    }
}
```
The generated graph is here:
<img width="694" alt="Screen Shot 2022-04-07 at 5 57 09 PM" src="https://user-images.githubusercontent.com/44760135/162326699-08b59ee7-e216-4692-a567-1b43b11996d1.png">

We should expect no `this.f` in both stores in this example.

**Comment by AndrewShf** [comment]
<img width="525" alt="image" src="https://user-images.githubusercontent.com/44760135/162326909-52d3c4bb-9e92-46f8-b5e8-b27ef5c9689e.png">
Maybe add the logic (determine if it is static method or not) in the `if (!isConstructor)` branch is a proper way of fixing this? @wmdietl

**Comment by wmdietl** [comment]
Thanks for reproducing this issue!
Instead of images, a permalink to the code would be easier - it took a second to see which class this is from.

It is https://github.com/eisop/checker-framework/blob/567084f11278273b33289d138d9d2a1b513d3c35/framework/src/main/java/org/checkerframework/framework/flow/CFAbstractTransfer.java#L391

Try introducing a new variable to store whether `methodTree` is static and then in the loop, if the method is static and the `varEle` is not static, you simply `continue` as you don't need to do anything.

**Comment by wmdietl** [comment]
Fixed by #216.
